framework/py/flwr/cli/auth_plugin/noop_auth_plugin.py

```
@@ -0,0 +1,69 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Concrete NoOp implementation for CLI-side account authentication plugin."""
+
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Optional, Union
+
+import typer
+
+from flwr.common.typing import UserAuthCredentials, UserAuthLoginDetails
+from flwr.proto.control_pb2_grpc import ControlStub
+
+from .auth_plugin import CliAuthPlugin
+
+
+class NoOpCliAuthPlugin(CliAuthPlugin):
+    """No-operation implementation of the CliAuthPlugin."""
+
+    @staticmethod
+    def login(
+        login_details: UserAuthLoginDetails,
+        control_stub: ControlStub,
+    ) -> UserAuthCredentials:
+        """Warn that no authentication is being performed and return empty tokens."""
+        # Warn the account that no authentication is being performed
+        typer.secho(
+            "Warning: No authentication is being performed. "
+            "Proceeding without account authentication.",
+            fg=typer.colors.YELLOW,
+        )
+        return UserAuthCredentials(
+            access_token="",
+            refresh_token="",
+        )
```

Instead of returning an empty object here, should we rather raise an error?

---

That will break `flwr login` and result in a long traceback. Wdyt? We can also raise a `typer.Exit` here  to avoid the long traceback if you prefer.

---

Hm... I'm not a fan of using `typer` at this level. Could we raise an exception, catch it on the caller side and use `typer.Exit` there?

---

Yeah, we can. I can also adjust our `OidcCliAuthPlugin` to raise an exception if fails.